File: backend/python-api/app/service/settings_service.py
from __future__ import annotations

from app.domain.app_settings import AppSettings
from app.repository.settings_repo import SettingsRepository


class SettingsService:
    def __init__(self, settings_repo: SettingsRepository):
        self._settings = settings_repo

    async def get_all(self) -> list[AppSettings]:
        return await self._settings.get_all()

    async def get(self, key: str) -> str:
        return await self._settings.get(key)

    async def set(self, key: str, value: str, session) -> None:
        await self._settings.set(key, value)
        await session.commit()

File: templates/backends/go-gin/internal/service/settings_service.go
package service

import (
	"context"

	"saas-starter/backend/go-api/internal/domain"
	"saas-starter/backend/go-api/internal/repository"
)

type SettingsService struct {
	settings repository.SettingsRepository
}

func NewSettingsService(settings repository.SettingsRepository) *SettingsService {
	return &SettingsService{settings: settings}
}

func (s *SettingsService) GetAll(ctx context.Context) ([]domain.AppSettings, error) {
	return s.settings.GetAll(ctx)
}

func (s *SettingsService) Get(ctx context.Context, key string) string {
	return s.settings.Get(ctx, key, "")
}

func (s *SettingsService) Set(ctx context.Context, key, value string) error {
	return s.settings.Set(ctx, key, value)
}
